### DISCUSSION TITLE
How do we identify conserved cell type markers after SCTransform - IntegrateLayers in Seurat 5?

### DISCUSSION THREAD

**Discussion by greengeckoruns** [satijalab/seurat#8740] [Q&A]
Hello, and thanks to everyone in advance for reading this question. I use Seurat 5 to analyze a single-cell experiment with two conditions (A vs. B). At the moment, I am trying out different data (pre)processing steps (NormalizeData-FindVariableFeatures-ScaleData [NFS] vs. SCTransform [SCT]) and integrations (cca, rpca, fastnmm, harmony, sci) to learn how different approaches influence the interpretation of results. My starting Seurat object has 2 layers (counts.A and counts.B), and the code for the NFS processing pipeline looks something like this:

```
obj_NFS <- NormalizeData(obj_NFS)
obj_NFS <- FindVariableFeatures(obj_NFS)
obj_NFS <- ScaleData(obj_NFS)
obj_NFS <- RunPCA(obj_NFS)
obj_NFS <- IntegrateLayers(object = obj_NFS, method = CCAIntegration, orig.reduction = "pca", new.reduction = "nfs.integrated.cca")
obj_NFS <- FindNeighbors(obj_NFS, dims = 1:30, reduction = "nfs.integrated.cca")
obj_NFS <- FindClusters(obj_NFS, resolution = 0.4, cluster.name = "nfs_cca_clusters")
obj_NFS <- RunUMAP(obj_NFS, dims = 1:30, reduction = "nfs.integrated.cca", reduction.name = "umap.nfs.cca")
obj_NFS <- JoinLayers(obj_NFS)
cluster1_markers <- FindConservedMarkers(obj_NFS, ident.1 = 1, grouping.var = "orig.ident")
```
The above works nicely irrespective of the integration method used (cca, rpca, fastnmm, harmony, scvi) during the IntegrateLayers() function call. However, I am having issues using FindConservedMarkers() on the SCT processed Seurat object. The code is the following:

```
obj_SCT <- SCTransform(obj_SCT)
obj_SCT <- RunPCA(obj_SCT)
obj_SCT <- IntegrateLayers(object = obj_SCT, method = CCAIntegration, orig.reduction = "pca", normalization.method = "SCT", new.reduction = "sct.integrated.cca")
obj_SCT <- FindNeighbors(obj_SCT, dims = 1:30, reduction = "sct.integrated.cca")
obj_SCT <- FindClusters(obj_SCT, resolution = 0.4, cluster.name = "sct_cca_clusters")
obj_SCT <- RunUMAP(obj_SCT, dims = 1:30, reduction = "sct.integrated.cca", reduction.name = "umap.sct.cca")
```
However, when I now try to run:

`FindConservedMarkers(obj_SCT, ident.1 = 1, grouping.var = "orig.ident")`

I get the following error, irrespective of whether the PrepSCTFindMarkers(obj_SCT) has been called or not:

```
Testing group A: (1) vs (0, 9, 3, 4, 5, 7, 10, 6, 11, 8, 2)
Warning: No layers found matching search pattern provided
Error in h(simpleError(msg, call)) : 
  error in evaluating the argument 'x' in selecting a method for function 'rowSums': subscript out of bounds
In addition: Warning messages:
1: Layer 'data' is empty 
2: Layer 'data' is empty
```

On the other hand, `FindMarkers(obj_SCT, ident.1 = 1, ident.2 = 5)` runs as expected after calling PrepSCTFindMarkers(obj_SCT).

I would appreciate any suggestions on what I am doing wrong. I also apologize if I am missing something obvious, and thanks for reading this post/helping.

**Comment by dr-dvice** [comment]
I also would like some clarification on how SCTransform should be used with the integrative analysis pipeline.
I also run into the same error when using SCTransform() on my two conditions and then trying SelectIntegrationFeatures()

**Reply by shaemclaughlin** [reply]
Hello, I am having this same issue in October 2025 - has anyone posted a solution or explanation to this error?
